pkg/cache/internal/cached_resource.go

```diff
@@ -113,6 +124,11 @@ func (c *CachedResource) HasTTL() bool {
 	return c.ttl != nil
 }
 
+// IsOnDemandOnly returns whether the resource is on-demand only.
```

Comment:
As this is internal, maybe make the comment a bit more specific (also maybe just `OnDemandOnly`)
```
OnDemandOnly returns whether the resource should be ignored by wildcard watches if not explicitly requested.
```